tools/stronghold/src/bc_linter.py

```diff
@@ -0,0 +1,22 @@
+"""Utilities for marking API compatibility checks."""
+
+from __future__ import annotations
+
+from typing import Callable, TypeVar, Any
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+def check_compat(*, enable: bool = True) -> Callable[[F], F]:
+    """Decorator used by stronghold to toggle API compatibility checks.
+
+    When ``enable`` is ``False`` the decorated function will be skipped by the
+    backward compatibility linter.
+    """
+
+    def decorator(func: F) -> F:
```

Comment:
If I'm understanding the intention here correctly, this decorator should be defined in the client code (unless we require bc-linter clients to import its codebase in runtime, which we should not!). The current implementation appears to support this (i.e. any decorator in VLLM codebase that sets `"_bc_linter_enable" will be recognized by bc-linter), however, the way how the code is structured makes it seem like `@check_compat` is the "official" annotation that needs to be used.

I'd suggest to change this:
1. move `check_compat` definition into test files
2. document the way to do `setattr(func, "_bc_linter_enable", enable)` in the client code instead (maybe pointing to `check_compat` as an example)